tests/lapi/builtin_select_test.lua

```diff
@@ -0,0 +1,48 @@
+--[[
+SPDX-License-Identifier: ISC
+Copyright (c) 2023-2025, Sergey Bronnikov.
+
+5.1 – Basic Functions
+https://www.lua.org/manual/5.1/manual.html
+
+Synopsis: select(index, ...)
+]]
+
+local luzer = require("luzer")
+local test_lib = require("lib")
+
+local unpack = unpack or table.unpack
+local MAX_INT = test_lib.MAX_INT
+local MIN_INT = test_lib.MIN_INT
+local MAX_STR_LEN = test_lib.MAX_STR_LEN
+
+local function TestOneInput(buf)
+    local fdp = luzer.FuzzedDataProvider(buf)
+
+    local items = fdp:consume_integer(1, 100)
+    local items_type = fdp:oneof({"number", "integer", "boolean", "string"})
+    local items_table
+    if items_type == "string" then
+        items_table = fdp:consume_strings(MAX_STR_LEN, items)
+    elseif items_type == "boolean" then
+        items_table = fdp:consume_booleans(items)
+    elseif items_type == "integer" then
+        items_table = fdp:consume_integers(MIN_INT, MAX_INT, items)
+    elseif items_type == "number" then
+        items_table = fdp:consume_numbers(MIN_INT, MAX_INT, items)
+    else
+        assert("Unsupported type")
+    end
+
+    local index = fdp:consume_boolean() and "#" or
+                  fdp:consume_integer(0, MAX_INT)
+    if index == "#" then
+        assert(select(index, unpack(items_table)) == items)
+    end
+    pcall(select, index, items_table)
```

Comment:
`select()` also supports formats like the following:
```
select(idx, ...)
select(idx, a, b, c)
select(idx, a, b, c, ...)
```

It is better to generate these cases and load them via `loadstring()` as well.

Reply:
```diff
 
+    local is_table = fdp:consume_boolean()
     local index = fdp:consume_boolean() and "#" or
                   fdp:consume_integer(0, MAX_INT)
     if index == "#" then
-        assert(select(index, unpack(items_table)) == items)
+        assert(select(index, is_table and tbl or unpack(tbl)) == count)
     end
-    pcall(select, index, items_table)
+    pcall(select, index, is_table and tbl or unpack(tbl))
 end
 
 local args = {
```